src/main/java/org/beanpod/switchboard/dao/StreamDaoImpl.java

```diff
@@ -58,4 +71,25 @@ public void updateStream(StreamDTO streamDto) {
     StreamEntity streamEntity = mapper.toEntity(streamDto);
     streamRepository.save(streamEntity);
   }
+
+  private boolean onSamePrivateNetwork(
+      InputChannelDTO inputChannelDto, OutputChannelDTO outputChannelDto) {
+    return inputChannelDto
+        .getDecoder()
+        .getDevice()
+        .getPublicIpAddress()
+        .equals(outputChannelDto.getEncoder().getDevice().getPublicIpAddress());
+  }
+
+  private boolean deviceOnLocalNetwork(DeviceDTO deviceDto) {
+    return deviceDto.getPublicIpAddress().equals(deviceDto.getPrivateIpAddress())
+        || deviceDto.getPublicIpAddress().equals(LOOPBACK_IP_V4)
+        || deviceDto.getPublicIpAddress().equals(LOOPBACK_IP_V6);
+  }
+
+  private boolean onLocalNetwork(
+      InputChannelDTO inputChannelDTO, OutputChannelDTO outputChannelDTO) {
+    return deviceOnLocalNetwork(inputChannelDTO.getDecoder().getDevice())
+        && deviceOnLocalNetwork(outputChannelDTO.getEncoder().getDevice());
+  }
```

Comment:
these three methods seem like they dont belong in the dao, more in a service package or a utility package

Reply:
Great suggestion, I moved them to `NetworkingUtil`. Much, much easier to test that way also.